cpp/src/gandiva/gdv_function_stubs.cc

```diff
@@ -611,6 +611,67 @@ int32_t gdv_fn_cast_intervalyear_utf8_int32(int64_t context_ptr, int64_t holder_
   auto* holder = reinterpret_cast<gandiva::IntervalYearsHolder*>(holder_ptr);
   return (*holder)(context, data, data_len, in1_validity, out_valid);
 }
+
+GANDIVA_EXPORT
+gdv_timestamp to_utc_timezone_timestamp(int64_t context, gdv_timestamp time_miliseconds,
+                                        const char* timezone, gdv_int32 length) {
+  using std::chrono::milliseconds;
+  using arrow_vendored::date::sys_time;
+  using arrow_vendored::date::time_zone;
+  using arrow_vendored::date::zoned_time;
+  using arrow_vendored::date::locate_zone;
+
+  sys_time <milliseconds> tp {milliseconds{time_miliseconds}};
+  try {
+    const time_zone* local_tz = locate_zone(std::string(timezone, length));
+    gdv_timestamp offset = local_tz->get_info(tp).offset.count()*1000;
+    return time_miliseconds - static_cast<gdv_timestamp>(offset);
+  } catch(...) {
+    int32_t msg_len = static_cast<int32_t>(strlen(timezone) + 50);
+    char* err_msg = reinterpret_cast<char*>(gdv_fn_context_arena_malloc(context,
+                                            msg_len));
+    if (err_msg == nullptr) {
+      gdv_fn_context_set_error_msg(context, "Could not allocate memory");
+      return 0;
+    }
+    std::snprintf(err_msg, msg_len, "'%s' is an invalid time zone name.", timezone);
+    gdv_fn_context_set_error_msg(context, err_msg);
```

Comment:
@anthonylouisbsb Could you confirm whether this is a preferred error message generation approach in Gandiva? (I'm not familiar with Gandiva's error message generation.)